### ISSUE TITLE
[BUG]: ML example not working with latest ML.NET version

### ISSUE THREAD

**Issue by indy-3rdman** [dotnet/spark#732]
**Describe the bug**

I don't think
https://github.com/dotnet/spark/tree/master/examples/Microsoft.Spark.CSharp.Examples/MachineLearning/Sentiment
is working correctly with the latest ML.NET version.

**To Reproduce**

1. Create a new solution in Visual Studio following the instructions.
2. Add the references to the latest Microsoft.Spark, Microsoft.ML and Microsoft.ML.DataView NuGet packages
3. Send to spark via spark-submit
4. See error "org.apache.spark.api.python.PythonException: System.TypeLoadException: Could not load type 'Microsoft.ML.Data.DataViewTypeAttribute' from assembly 'Microsoft.ML.DataView, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'"

5. Manually copying the latest Microsoft.ML.DataView.dll into DOTNET_WORKER_DIR solves the problem.

**Expected behavior**

Ideally the DLLs/versions used in the folder of the .NET for Apache Spark application should be used.
Alternatively it might be worth mentioning this behavior and the workaround in the documentation.

**Screenshots**

![ml-dependencies](https://user-images.githubusercontent.com/60756651/95659258-18ed5c80-0b20-11eb-9ad8-c9e9b82b9e27.png)

**Comment by WalternativE** [comment]
I can reproduce this using F# interactive (.NET 5-RC2).

This would be the failing script in its entirety
```fsharp
#r "nuget: Microsoft.ML, 1.5.2"
#r "nuget: Microsoft.ML.DataView, 1.5.2"

open Microsoft.ML
open Microsoft.ML.Data

type House =
    { Size: float
      Price: float }

type Prediction =
    { [<ColumnName("Score")>]Price: float }

let mlContext = MLContext()

let houseData = [
   { Size = 1.1; Price = 1.2 }
   { Size = 1.9; Price = 2.3 }
   { Size = 2.8; Price = 3.0 }
   { Size = 3.4; Price = 3.7 }
]

let trainingData = mlContext.Data.LoadFromEnumerable(houseData)
```
The execution of the last line fails.

If I downgrade both libraries to version `1.0.0` it works. 

The output of `dotnet --info` on my system is
```bash
.NET SDK (reflecting any global.json):
 Version:   5.0.100-rc.2.20479.15
 Commit:    da7dfa8840

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  20.04
 OS Platform: Linux
 RID:         ubuntu.20.04-x64
 Base Path:   /usr/share/dotnet/sdk/5.0.100-rc.2.20479.15/

Host (useful for support):
  Version: 5.0.0-rc.2.20475.5
  Commit:  c5a3f49c88

.NET SDKs installed:
  3.1.403 [/usr/share/dotnet/sdk]
  5.0.100-rc.2.20479.15 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.9 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.0-rc.2.20475.17 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.9 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.0-rc.2.20475.5 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
```

**Comment by rahulshivangi** [comment]
Hi, 


I am facing the same issue, getting below exception:
`System.TypeLoadException: Could not load type 'Microsoft.ML.Data.DataViewTypeAttribute' from assembly 'Microsoft.ML.DataView, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'`

I am using Microsoft.Spark v 1.0.0 on **Azure Synapze**
I am new to Azure Spark, Please excuse me, if I misunderstood anything.
Do you know any workaround that I can use while submitting jobs on Azure Synapse?

Thanks,

**Comment by Niharikadutta** [comment]
Resolved with latest V2.1.0 release. Closing this issue.
